### ISSUE TITLE
Brave-added Content Settings Types not auto-generated for Android

### ISSUE THREAD

**Issue by ShivanKaul** [brave/brave-browser#28838]
The `chromium_src` overrides for the [Content Settings Types](https://github.com/brave/brave-core/blob/master/chromium_src/components/content_settings/core/common/content_settings_types.h) C++ enum don't get auto-generated in Java (verify by checking the locally generated `ContentSettingsType.class` to see that none of the Brave_* enums exist in that file.) 

This is a problem because it prevents porting new Desktop site settings to Android (accessing them in Java leads to a build error). 

The underlying issue is that `build/android/gyp/java_cpp_enum.py` which generates the Java enum counterpart runs without the Brave overrides. We could potentially use `build/android/gyp/gcc_preprocess.py` to preprocess every file. Need to investigate.

**Comment by ShivanKaul** [comment]
cc @bridiver @wchen342

**Comment by ShivanKaul** [comment]
I looked into this a bit. I don't think preprocessing the header file is a good idea, or at least the effort  will not be worth it:
1. We can't call gcc_preprocess.py directly, because it needs a bunch of arguments (template, locales etc) that we don't have/need in java_cpp_enum.py.
2. We'll need to call the compiler directly in java_cpp_enum.py to preprocess the files that have a chromium_src counterpart (i.e. which we're overriding), but this got tricky when I tried it -- it needs a bunch of directory includes to make sure we're capturing all possible includes done in our overriding header files. For e.g., `chromium_src/components/content_settings/core/browser/cookie_settings.h` will need `-I../third_party/ffmpeg/libavcodec/` to account for `#include <jni.h>` in `/components/content_settings/core/browser/content_settings_rule.h:15`, which is in the include chain for cookie_settings.h. There are many others like this, and any hardcoded list of directories we come up with here will be brittle. 
3. Even if we only preprocess the one header file we care about here (content_settings_types.h), we'll have to modify the regex at https://source.chromium.org/chromium/chromium/src/+/main:build/android/gyp/java_cpp_enum.py;l=175 in order to account for the fact that we have multiple enum values on the same line. From past experience with brittle regexes, I think this would be an awful idea :) 

All of this will be a messy patch on java_cpp_enum.py. So IMO we're better off just having a direct patch on `chromium_src/components/content_settings/core/common/content_settings_types.h` for the few enums we want on Android as well.
